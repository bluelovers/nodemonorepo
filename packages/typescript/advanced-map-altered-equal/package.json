{
  "name": "advanced-map-altered-equal",
  "version": "0.0.3",
  "description": "Map-like class with customizable equality operator",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodemodules/nodemonorepo.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "keywords": [
    "map",
    "equal",
    "custom"
  ],
  "dependencies": {
    "tslib": "^1.9.3",
    "convenient-typescript-utilities": "^0.0.3",
    "advanced-map-base": "^0.0.2",
    "advanced-map-types": "^0.0.2",
    "@types/node": "^11.12.0"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
