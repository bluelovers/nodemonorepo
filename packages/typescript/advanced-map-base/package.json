{
  "name": "advanced-map-base",
  "version": "0.0.0",
  "description": "Abstract base class of advanced-map packages",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodemodules/nodemonorepo.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "keywords": [
    "map",
    "abstract",
    "base",
    "class"
  ],
  "dependencies": {
    "tslib": "^1.9.3",
    "typescript-miscellaneous": "^0.1.0",
    "advanced-map-types": "^0.0.0",
    "@types/node": "^10.9.3"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
