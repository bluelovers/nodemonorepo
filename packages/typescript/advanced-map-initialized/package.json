{
  "name": "advanced-map-initialized",
  "version": "0.0.0",
  "description": "Map-like class with default values",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodemodules/nodemonorepo.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "keywords": [
    "map",
    "default",
    "initialize",
    "value"
  ],
  "dependencies": {
    "tslib": "^1.9.3",
    "advanced-map-base": "^0.0.1",
    "advanced-map-types": "^0.0.1",
    "@types/node": "^10.9.3"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
