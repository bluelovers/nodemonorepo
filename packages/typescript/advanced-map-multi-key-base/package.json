{
  "name": "advanced-map-multi-key-base",
  "version": "0.0.1",
  "description": "Abstract base class of advanced-map-multi-key and advanced-map-mutable-multi-key",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodemodules/nodemonorepo.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "dependencies": {
    "tslib": "^1.9.3",
    "create-array-equal": "^0.0.0",
    "advanced-map-altered-equal": "^0.0.2",
    "advanced-map-types": "^0.0.2",
    "@types/node": "^10.9.3"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
