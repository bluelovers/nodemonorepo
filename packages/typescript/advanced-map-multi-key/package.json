{
  "name": "advanced-map-multi-key",
  "version": "0.0.2",
  "description": "Map-like class using multi keys",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodemodules/nodemonorepo.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "keywords": [
    "map",
    "multiple",
    "keys"
  ],
  "dependencies": {
    "iter-tools": "^6.1.4",
    "tslib": "^1.9.3",
    "advanced-map-altered-equal": "^0.0.1",
    "advanced-map-types": "^0.0.1",
    "@types/node": "^10.9.3"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
