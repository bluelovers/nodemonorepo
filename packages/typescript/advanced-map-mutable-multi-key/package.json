{
  "name": "advanced-map-mutable-multi-key",
  "version": "0.0.3",
  "description": "Map-like class using mutable arrays as keys",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodemodules/nodemonorepo.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "keywords": [
    "map",
    "multiple",
    "keys",
    "mutable"
  ],
  "dependencies": {
    "tslib": "^1.9.3",
    "advanced-map-multi-key-base": "^0.0.3",
    "advanced-map-types": "^0.0.2",
    "@types/node": "^10.9.3"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
