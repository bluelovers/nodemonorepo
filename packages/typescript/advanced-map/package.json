{
  "name": "advanced-map",
  "version": "0.0.3",
  "description": "Advanced Map-like classes",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodemodules/nodemonorepo.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "keywords": [
    "map",
    "advanced",
    "meta"
  ],
  "dependencies": {
    "tslib": "^1.9.3",
    "advanced-map-types": "^0.0.1",
    "advanced-map-base": "^0.0.1",
    "advanced-map-multi-key-base": "^0.0.0",
    "advanced-map-altered-equal": "^0.0.1",
    "advanced-map-multi-key": "^0.0.2",
    "advanced-map-mutable-multi-key": "^0.0.0",
    "advanced-map-initialized": "^0.0.0",
    "@types/node": "^10.9.3"
  },
  "devDependencies": {
    "iter-tools": "^6.1.4"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
