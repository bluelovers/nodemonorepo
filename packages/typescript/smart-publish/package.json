{
  "name": "smart-publish",
  "version": "0.0.3",
  "description": "Like npm publish, but also tag prepreleases and knows when it is public",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "keywords": [
    "publish",
    "prerelease",
    "tag",
    "npm"
  ],
  "bin": {
    "smart-publish": "./bin/smart-publish"
  },
  "dependencies": {
    "chalk": "^2.4.2",
    "semver": "^6.0.0",
    "tslib": "^1.9.3",
    "exec-inline": "^0.0.2",
    "@types/semver": "^5.5.0",
    "@types/node": "^11.12.0"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
