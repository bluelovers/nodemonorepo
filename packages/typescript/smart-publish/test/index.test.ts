import path from 'path'

import {
  cmds,
  tags,
  bin,
  ExitStatusCode
} from '../index'

describe('bin', () => {
  it('matches package.json', () => {
    expect(bin).toBe(path.resolve(
      __dirname,
      '..',
      require('../package.json').bin['smart-publish']
    ))
  })
})

describe('ExitStatusCode', () => {
  it('matches snapshot', () => {
    expect(ExitStatusCode).toMatchSnapshot()
  })
})

describe('cmds', () => {
  const name = '<PackageName>'

  it('when version is invalid', () => {
    expect(() => cmds(name, '<InvalidVersion>')).toThrowErrorMatchingSnapshot()
  })

  it('when version has no suffix', () => {
    const result = cmds(name, '0.1.2')
    expect(result).toEqual([['publish', '--access', 'public']])
  })

  it('when version has a non-text suffix', () => {
    const result = cmds(name, '0.1.2-3')
    expect(result).toEqual([['publish', '--tag', 'prerelease', '--access', 'public']])
  })

  it('when version has a text suffix', () => {
    const result = cmds(name, '0.1.2-rc0')
    expect(result).toMatchSnapshot()
  })

  it('when version has a suffix of exactly 0', () => {
    const result = cmds(name, '0.1.2-0')
    expect(result).toEqual([['publish', '--tag', 'prerelease', '--access', 'public']])
  })
})

describe('tags', () => {
  it('when suffix is not provided', () => {
    expect(tags()).toEqual(new Set())
  })

  it('when suffix is non-text', () => {
    expect(tags('123')).toEqual(new Set(['prerelease']))
  })

  it('when suffix is text', () => {
    expect(tags('rc0')).toEqual(new Set(['rc', 'prerelease']))
  })

  it('when suffix is exactly 0', () => {
    expect(tags('0')).toEqual(new Set(['prerelease']))
  })

})
