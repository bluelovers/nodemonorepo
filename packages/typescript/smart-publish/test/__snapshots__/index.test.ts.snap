// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExitStatusCode matches snapshot 1`] = `
Object {
  "0": "Success",
  "1": "Other",
  "2": "MissingName",
  "3": "InvalidVersion",
  "InvalidVersion": 3,
  "MissingName": 2,
  "Other": 1,
  "Private": 1,
  "Success": 0,
}
`;

exports[`cmds when version has a text suffix 1`] = `
Array [
  Array [
    "publish",
    "--tag",
    "rc",
  ],
  Array [
    "dist-tag",
    "add",
    "<PackageName>@0.1.2-rc0",
    "prerelease",
  ],
]
`;

exports[`cmds when version is invalid 1`] = `"Cannot parse \\"<InvalidVersion>\\" in semver format"`;
